rust/utils/hex/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+use std::error::Error;
```

Comment:
Honestly pretty uncomfortable with a lot of this file - could you beef it up with comments on exactly what's going on throughout and what the motivation is?
I find it pretty hard to read and want to make sure that by having something like this over some of the default functionality in ethers-rs we're able to revisit it / understand it at a glance in the future

Reply:
Added some more docs, yeah, was a little spur of the moment.